stories/Thread/102-ThreadPreview.stories.tsx

```diff
@@ -0,0 +1,36 @@
+import HiddenThread, { HiddenThreadProps } from "thread/ThreadPreview";
+
+import type { Meta, Story/*, StoryObj*/ } from "@storybook/react";
+
+export default {
+  title: "ThreadPreview / Hidden Thread",
+  component: HiddenThread,
+};
+
+const mockUnhide =  ({ 
```

Comment:
You can use `action` from storybook here. See this as an example: https://github.com/BobaBoard/boba-components/blob/main/stories/Buttons/400-PopupButtons.stories.tsx#L47

Here's the documentation: https://storybook.js.org/docs/angular/essentials/actions

Reply:
I'm not sure how to use actions here. I guess it's two things:
1. How do I reconcile the `action` object with the type checking in `HiddenThreadProps`?
2. I changed `HiddenThread` to use `ActionLink`, as you suggested in a comment below. Will the action callback work properly if it's passed to it?

Reply:
`action("action_name")` returns a function that you can pass wherever a function is required. You can think of it a bit as passing `console.log` (without parentheses) as a function argument so that the value passed to the function is logged when the function is called. 

So here you should be able to do:

```ts
export const Default = ThreadPreviewTemplate.bind({});
Default.args = {
  hidden: true,
  onThreadHidden: action("threadHidden"),
};
```

and that should be roughly equivalent to the `console.log` you have now (except that the log is going to be in storybook).

As for `ActionLink`, as I mentioned that's the wrong abstraction here, but let me clarify how that would work.

If you were using `ActionLink`, this would be the correct Props type:

```ts
export interface HiddenThreadProps {
  hidden: boolean;
  onThreadHidden: LinkWithAction;
}
```

and then your story could use action like:

```ts

Default.args = {
  hidden: true,
  onThreadHidden: {
    onClick: action("threadHidden"),
    href: "a_sample_href"
  },
};
```

The code you need to use action link is different from what you were doing in this PR, and indeed you probably got confused cause we couldn't use `ActionLink` here. As I said, there's no way to pass the `hide` argument to it. Sorry again for misleading you :)

Reply:
I got it to work via `import { action } from "@storybook/addons-action"`, as described [here](https://github.com/storybookjs/storybook/blob/master/addons/actions/ADVANCED.md)! Maybe there's a way to get it to work with the newer way of using actions, but I wouldn't know.

Reply:
I kind of got `LinkWithAction` to work, by creating it on the component side and passing in what the frontend sent. Like this:
```  
  let link : LinkWithAction = { 
    onClick: () => onThreadHidden(!hidden), 
  }
  return (
    <div className="post hidden">
      This thread was hidden{" "}
      <ActionLink link={link} allowDefault={false}>
        ...
```
That being said, it's not difficult to change it to what you suggested, if this is some unholy abomination against good practices 😛